### DISCUSSION TITLE
Is it possible to differentiate multiple cmdrs i nthe journal?

### DISCUSSION THREAD

**Discussion by netchild** [aussig/BGS-Tally#180] [Ideas]
Is it possible to see different cmdrs in the log, or is this a limiation of EMDC?

Case:
 - 2 different accounts on the same machine logging into the same journal files
 - 1 account does X
 - other account does Y

Currently X+Y are reported in the same tally, cmdrs are not differentiated. Would it be possible to separate the actions based upon the account which did the action?

**Comment by aussig** [comment]
Yes this is technically possible. 

However, this would be a very big change, as each CMDR would not just need its own current tally tracking, but also its own set of historical tallies, discord webhooks, API configurations, and settings.

There would need to be a “combined” mode for those players who don’t want tallies separate (to be consistent with the way it currently works) and a “separate” mode, but also probably a way to manually switch CMDR to check (and post) your work if for example you forgot to post, without requiring the player to reload the whole game just to switch CMDR in BGS-Tally. 

All of this would need to be implemented with an extremely clear user interface to show which CMDR you are working with - probably ok on the main window but how, for example, would that work on the settings screen? 

I think you can probably guess that all this is very unlikely to happen in the near future. Also I’m not sure there is great demand - this is only the second time this has been requested (though I concede that is no guide for how many silently want the feature).

**Comment by netchild** [comment]
In our discord with multiple people (~200 staff, 10000 members) with multiple cmdrs. Not all would use it, but I would guess about 20 would.

I would suggest a "by default combined" mode. And for the separate mode simply double the lines for the factions and add "(cmdr name)". An user can then chose what to report. If it is not used by alot of people, you will not get a lot of feature requests to improve this (the proper UI you talked about).

**Comment by aussig** [comment]
But presumably you would want each CMDR to post to different discord channels, otherwise I'm not sure I understand what the use case is (BGS work is BGS work, irrespective of who does it).

**Comment by netchild** [comment]
There is a difference if one cmdr is doing 2*Xmil bounties (applies also for other stuff), or two cmdrs are doing Xmil each (diminishing returns per cmdr once a limit is reached per cmdr). We have some people with 10 accounts.

I understand how you can imaing it to be useful to have different cmdrs send to different discord channels. For us we would need different discord channels depending on the system we operate in, but that is not something I want to make a feature request for. I do not need different discord channels for different cmdrs. What your plugin helps a lot, is to count what we do. By having it split by cmdr (sending to the same channel), I can at least decide to send all to the same channel, or only calculate only for one cmdr (if you would implement that feature) and copy&paste to discord. Personally I do not mind to configure a different discord channel by hand, I rarely have the need to send to more than one channel.

As I said, nothing complicated initially (repeat the factions names and add the cmdr name for each faction/cmdr), keep the rest as is, make it the default to have the combined calculation without adding a cmdr name. And then see if this is a feature which is useful to a lot of people and it would be beneficial to extend it, or not.

**Comment by aussig** [comment]
Ok, I'll have a think about this as it will require a restructure of the internal tracking data even for a simple approach.

**Comment by netchild** [comment]
Thanks for at least considering / thinking about it. Much appreciated!
